From: Takashi Iwai <tiwai@suse.de>
Subject: kABI workaround for snd_seq_kernel_client_enqueue() API changes
Patch-mainline: Never, ugly kABI workaround
References: bsc#1051510

The snd_seq_kernel_client_enqueue() API function was changed to take
fewer arguments.  Rename it and provide the old API functions to keep
the kABI compatibility with SLE15.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/sound/seq_kernel.h     |    2 ++
 sound/core/seq/seq_clientmgr.c |   17 +++++++++++++++++
 2 files changed, 19 insertions(+)

--- a/include/sound/seq_kernel.h
+++ b/include/sound/seq_kernel.h
@@ -73,6 +73,8 @@ __printf(3, 4)
 int snd_seq_create_kernel_client(struct snd_card *card, int client_index,
 				 const char *name_fmt, ...);
 int snd_seq_delete_kernel_client(int client);
+/* XXX kABI compatibility for SLE15 XXX */
+#define snd_seq_kernel_client_enqueue	__snd_seq_kernel_client_enqueue
 int snd_seq_kernel_client_enqueue(int client, struct snd_seq_event *ev,
 				  struct file *file, bool blocking);
 int snd_seq_kernel_client_dispatch(int client, struct snd_seq_event *ev, int atomic, int hop);
--- a/sound/core/seq/seq_clientmgr.c
+++ b/sound/core/seq/seq_clientmgr.c
@@ -2557,3 +2557,20 @@ void __exit snd_sequencer_device_done(vo
 	snd_unregister_device(&seq_dev);
 	put_device(&seq_dev);
 }
+
+/* XXX kABI compatibility for SLE15 XXX */
+#undef snd_seq_kernel_client_enqueue
+int snd_seq_kernel_client_enqueue(int client, struct snd_seq_event *ev,
+				  int atomic, int hop)
+{
+	return __snd_seq_kernel_client_enqueue(client, ev, NULL, false);
+}
+EXPORT_SYMBOL(snd_seq_kernel_client_enqueue);
+
+int snd_seq_kernel_client_enqueue_blocking(int client, struct snd_seq_event *ev,
+					   struct file *file, int atomic,
+					   int hop)
+{
+	return __snd_seq_kernel_client_enqueue(client, ev, file, true);
+}
+EXPORT_SYMBOL(snd_seq_kernel_client_enqueue_blocking);
